### PR TITLE
chore: bump centraldashboard image to 1.9.0-rc.0

### LINKED ISSUE #91: Update `kubeflow-central-dashboard` ROCK
### Context

Each charm in Kubeflow has a ROCK (or a set of ROCKs) it is using. The process of upgrading ROCKs usually means going to the component's upstream repository Dockerfile, comparing current ROCK version against the one in the repository and adding the missing bits.

### What needs to get done

1. Clone the [upstream's](https://github.com/kubeflow/kubeflow) kubeflow repository.
2. Go to desired Dockerfile (first line in `rockcraft.yaml` should specify the full path).
3. Checkout to the branch or tag of the target version.
4. Checkout to the branch or tag of the version of the current ROCK.
5. Compare both files to spot the differences.

### Definition of Done

1. New ROCK is ready.
2. Tests are passing.
3. ROCK is published under `charmedkubeflow` Dockerhub account.

### PR DESCRIPTION
**Pull request by orfeas-k**
Diff between [1.8-branch](https://github.com/kubeflow/kubeflow/blob/v1.8-branch/components/centraldashboard/Dockerfile#L36) and [1.9.0-rc.0 tag](https://github.com/kubeflow/kubeflow/blob/v1.9.0-rc.0/components/centraldashboard/Dockerfile#L27) -> https://www.diffchecker.com/JaCMol7i/.

Note that this PR renames also the `name` version since in order to:
1. follow the upstream name as we do in most of our rocks
2. follow the convention where rock directory and rock name are the same

#### Testing
1. `usr/src/app` directory is tested by sanity tests
3. Check that image runs and logs the same with upstream image
```
# upstream image
 ╰─$ docker run --rm kubeflownotebookswg/centraldashboard:v1.9.0-rc.0                                        

> kubeflow-centraldashboard@0.0.2 start
> npm run serve


> kubeflow-centraldashboard@0.0.2 serve
> node dist/server.js

Initializing Kubernetes configuration
Unable to fetch Nodes Error: connect ECONNREFUSED 127.0.0.1:8080
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8080
}
Unable to fetch Application information: Error: connect ECONNREFUSED 127.0.0.1:8080
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8080
}
Unable to fetch Nodes Error: connect ECONNREFUSED 127.0.0.1:8080
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8080
}
"other" is not a supported platform for Metrics
Using Profiles service at http://profiles-kfam.kubeflow:8081/kfam
Server listening on port http://localhost:8082 (in production mode)

# rock
╰─$ docker run --rm kubeflow-central-dashboard:1.9.0
2024-06-10T07:15:24.561Z [pebble] Started daemon.
2024-06-10T07:15:24.568Z [pebble] POST /v1/services 4.42732ms 202
2024-06-10T07:15:24.572Z [pebble] Service "serve" starting: /usr/bin/npm start
2024-06-10T07:15:25.107Z [serve] 
2024-06-10T07:15:25.107Z [serve] > kubeflow-centraldashboard@0.0.2 start
2024-06-10T07:15:25.107Z [serve] > npm run serve
2024-06-10T07:15:25.107Z [serve] 
2024-06-10T07:15:25.576Z [pebble] GET /v1/changes/1/wait 1.006711635s 200
2024-06-10T07:15:25.576Z [pebble] Started default services with change 1.
2024-06-10T07:15:25.611Z [serve] 
2024-06-10T07:15:25.611Z [serve] > kubeflow-centraldashboard@0.0.2 serve
2024-06-10T07:15:25.611Z [serve] > node dist/server.js
2024-06-10T07:15:25.611Z [serve] 
2024-06-10T07:15:26.829Z [serve] Initializing Kubernetes configuration
2024-06-10T07:15:26.857Z [serve] Unable to fetch Nodes Error: connect ECONNREFUSED 127.0.0.1:8080
2024-06-10T07:15:26.857Z [serve]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1138:16) {
2024-06-10T07:15:26.857Z [serve]   errno: 'ECONNREFUSED',
2024-06-10T07:15:26.857Z [serve]   code: 'ECONNREFUSED',
2024-06-10T07:15:26.857Z [serve]   syscall: 'connect',
2024-06-10T07:15:26.857Z [serve]   address: '127.0.0.1',
2024-06-10T07:15:26.857Z [serve]   port: 8080
2024-06-10T07:15:26.857Z [serve] }
2024-06-10T07:15:26.858Z [serve] Unable to fetch Application information: Error: connect ECONNREFUSED 127.0.0.1:8080
2024-06-10T07:15:26.858Z [serve]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1138:16) {
2024-06-10T07:15:26.858Z [serve]   errno: 'ECONNREFUSED',
2024-06-10T07:15:26.858Z [serve]   code: 'ECONNREFUSED',
2024-06-10T07:15:26.858Z [serve]   syscall: 'connect',
2024-06-10T07:15:26.858Z [serve]   address: '127.0.0.1',
2024-06-10T07:15:26.858Z [serve]   port: 8080
2024-06-10T07:15:26.858Z [serve] }
2024-06-10T07:15:26.858Z [serve] Unable to fetch Nodes Error: connect ECONNREFUSED 127.0.0.1:8080
2024-06-10T07:15:26.858Z [serve]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1138:16) {
2024-06-10T07:15:26.858Z [serve]   errno: 'ECONNREFUSED',
2024-06-10T07:15:26.858Z [serve]   code: 'ECONNREFUSED',
2024-06-10T07:15:26.858Z [serve]   syscall: 'connect',
2024-06-10T07:15:26.858Z [serve]   address: '127.0.0.1',
2024-06-10T07:15:26.858Z [serve]   port: 8080
2024-06-10T07:15:26.858Z [serve] }
2024-06-10T07:15:26.859Z [serve] "other" is not a supported platform for Metrics
2024-06-10T07:15:26.859Z [serve] Using Profiles service at http://profiles-kfam.kubeflow:8081/kfam
2024-06-10T07:15:26.866Z [serve] Server listening on port http://localhost:8082 (in production mode)
```

Closes #91

### CODE REVIEW

File: centraldashboard/rockcraft.yaml
Comment (misohu) [inline centraldashboard/rockcraft.yaml:1]: Upstream uses not tiny https://github.com/krallin/tini to server the dashboard https://github.com/kubeflow/kubeflow/blob/v1.9.0-rc.0/components/centraldashboard/Dockerfile#L31

File: centraldashboard/rockcraft.yaml [hunk 1]
```diff
@@ -65,17 +65,15 @@ parts:
...
-      npm install && \
-      npm test && \
+      npm ci && \
       npm run build && \
       npm prune --production
 
       # install build artifacts
-      mkdir -p ${CRAFT_PART_INSTALL}/app
-      cp -r ${CRAFT_PART_SRC}/components/centraldashboard/* ${CRAFT_PART_INSTALL}/app
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/src/app
```
Comment (misohu) [inline centraldashboard/rockcraft.yaml:73]: Why are we changing the directory ?

**Review by misohu — COMMENTED** [review]
Added comments :)

File: centraldashboard/rockcraft.yaml (hunk 1 above)
Comment (orfeas-k) [reply centraldashboard/rockcraft.yaml:73]: To follow what [upstream](https://github.com/kubeflow/kubeflow/blob/v1.9.0-rc.0/components/centraldashboard/Dockerfile#L27) does. This way, we will need to add minimal changes whenever switching between upstream and rock.

File: centraldashboard/rockcraft.yaml
Comment (orfeas-k) [reply centraldashboard/rockcraft.yaml:1]: I don't think we need it ([thread in rockcraft](https://matrix.to/#/!NPPCseDHKRvSBMUEXN:ubuntu.com/$KbLXHxcP1dLMvR1G1nmQbFPl85iHle9Za-bjoDE8Dp8?via=ubuntu.com&via=matrix.org) and in our [private channel](https://matrix.to/#/!tWZKiDikyoQKvXPGge:ubuntu.com/$kVUbRCpvvOsURd7Zgd6PExHcXQgrqWDyFjlvlNpmFqE?via=ubuntu.com)). Tini is essentially being used to manage processes in a container while we have Pebble for that. Some functionalities, like signal rewriting, are not available in Pebble but I don't think we need those here. WDYT?

**Review by misohu — APPROVED** [review]
Good job :)

**Pull request merged**